test/test_torch.py

```diff
@@ -12303,6 +12309,24 @@ def non_zero_rand(size, dtype, device):
                     non_zero_rand((2, 2), dtype=dtype, device=device),
                     non_zero_rand((2, 2), dtype=dtype, device=device))
 
+    # This function tests that a nan value is returned for input values not in domain
+    def test_acosh_domain(self, device):
+        # Domain of acosh is [1, inf)
+        sample = torch.tensor([-1.23, -0.06, 0.98], device=device)
+        self.assertEqual(torch.isnan(torch.acosh(sample)), torch.BoolTensor([1, 1, 1]))
+        self.assertEqual(torch.isnan(sample.acosh()), torch.BoolTensor([1, 1, 1]))
+
+    # This function tests that a nan value is returned for input values not in domain
+    def test_atanh_domain(self, device):
```

Comment:
Same comments as above, except in this case I think your sample values are sufficient (but maybe toss -inf and inf in there for fun, anyway).

Reply:
Done! Verified the domain values from here: https://en.cppreference.com/w/cpp/numeric/math/atanh.